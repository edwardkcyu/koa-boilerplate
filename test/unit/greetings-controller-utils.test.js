const moment = require('moment');

const { formatDate } = require('../../src/controllers/utils');

describe('formatDate', () => {
  test('format date correctly', () => {
    const date = moment([2018, 0, 1, 11, 11, 11]);

    expect(formatDate(date)).toMatchSnapshot();
  });

  test('throw error if input is null', () => {
    const date = null;

    expect(() => formatDate(date)).toThrowErrorMatchingSnapshot();
  });
});
