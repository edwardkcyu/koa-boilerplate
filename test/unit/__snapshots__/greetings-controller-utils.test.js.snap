// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatDate format date correctly 1`] = `"2018-01-01 11:11:11"`;

exports[`formatDate throw error if input is null 1`] = `"date should not be null"`;
